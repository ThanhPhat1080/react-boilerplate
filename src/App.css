#root {
  width: 100%;
  height: 100%;
}

.App {
  text-align: center;
}

.logo {
  height: 6em;
  padding: 1.5em;
  will-change: filter;
}
.logo:hover {
  filter: drop-shadow(0 0 2em #646cffaa);
}
.logo.react:hover {
  filter: drop-shadow(0 0 2em #61dafbaa);
}
.logo.ts:hover {
  filter: drop-shadow(0 0 2em #ffffffaa);
}

@keyframes logo-rotate {
  from {
    transform: rotate(0deg);
  }
  to {
    transform: rotate(360deg);
  }
}

@keyframes logo-scale {
  0% {
    transform: scale(1.2);
  }
  50% {
    transform: scale(1);
  }
  100% {
    transform: scale(1.2);
  }
}

@keyframes logo-spin {
  0% {
    transform: translateX(-4px);
  }

  50% {
    transform: translateX(5px);
  }

  100% {
    transform: translateX(-4px);
  }
}

@media (prefers-reduced-motion: no-preference) {
  a:nth-of-type(1) .logo {
    animation: logo-scale infinite 1.5s linear;
  }

  a:nth-of-type(2) .logo {
    animation: logo-rotate infinite 10s linear;
  }

  a:nth-of-type(3) .logo {
    animation: logo-spin infinite 1s linear;
  }
}

.card {
  padding: 2em;
}

.read-the-docs {
  color: #888;
}
