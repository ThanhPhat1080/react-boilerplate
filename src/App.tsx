// Assets
import { ICONS } from '@assets';

// Components
import { BrowserRouter, Routes, Route } from 'react-router-dom';
import { HomePage, Zoo } from '@Pages';

// Enums
import {ROUTERS} from '@enums'

// Styles
import './App.css';

function App() {
  return (
    <BrowserRouter>
      <div className='App'>
        <div>
          <a href='https://vitejs.dev' target='_blank' rel='noreferrer'>
            <img src={ICONS.VITE_ICON} className='logo' alt='Vite logo' />
          </a>
          <a href='https://reactjs.org' target='_blank' rel='noreferrer'>
            <img src={ICONS.REACT_ICON} className='logo react' alt='React logo' />
          </a>

          <a href='https://www.typescriptlang.org/' target='_blank' rel='noreferrer'>
            <img src={ICONS.TS_ICON} className='logo ts' alt='TS logo' />
          </a>
        </div>
        <Routes>
          <Route path={ROUTERS.ROOT}>
            <Route index element={<HomePage />} />
            <Route path={ROUTERS.ZOO} element={<Zoo />} />
          </Route>
        </Routes>
      </div>
    </BrowserRouter>
  );
}

export default App;
