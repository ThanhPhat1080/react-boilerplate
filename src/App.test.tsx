// Libraries
import * as renderer from 'react-test-renderer';

// Components
import App from './App';

describe('Testing App', () => {
  test('It should render correctly', () => {
    const component = renderer
      .create(
          <App />
      )
      .toJSON();
    expect(component).toMatchSnapshot();
  });
});
