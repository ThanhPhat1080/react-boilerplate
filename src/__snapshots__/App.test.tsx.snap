// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing App It should render correctly 1`] = `
<div
  className="App"
>
  <div>
    <a
      href="https://vitejs.dev"
      rel="noreferrer"
      target="_blank"
    >
      <img
        alt="Vite logo"
        className="logo"
      />
    </a>
    <a
      href="https://reactjs.org"
      rel="noreferrer"
      target="_blank"
    >
      <img
        alt="React logo"
        className="logo react"
      />
    </a>
    <a
      href="https://www.typescriptlang.org/"
      rel="noreferrer"
      target="_blank"
    >
      <img
        alt="TS logo"
        className="logo ts"
      />
    </a>
  </div>
  <div>
    <h1>
      You are at
       
      <span
        aria-label="home"
        role="img"
      >
        🏠
      </span>
    </h1>
    <nav>
      <a
        href="/zoo"
        onClick={[Function]}
      >
        Go to the Zoo
         
        <span
          aria-label="tend"
          role="img"
        >
          🎪
        </span>
      </a>
    </nav>
  </div>
</div>
`;
